types/pino/index.d.ts

```diff
@@ -221,9 +221,15 @@ declare namespace P {
          */
         level?: LevelWithSilent | string;
         /**
-         * Outputs the level as a string instead of integer. Default: `false`.
+         * (DEPRECATED, use `formatters.level`) Outputs the level as a string instead of integer. Default: `false`.
          */
         useLevelLabels?: boolean;
+        /**
+         * Changes the shape of the log level: Default shape: { level: number }
+         */
+        formatters?: {
+            [key: string]: (level: string, value: number) => Record<string, string>;
+        };
```

Comment:
Based on the [`formatters` documentation](https://getpino.io/#/docs/api?id=formatters-object), this type appears to be wrong.

Here's what I would expect instead:
```suggestion
        /**
         * An object containing functions for formatting the shape of the log lines. These functions should return a
         * JSONifiable object and should never throw. These functions allow for full customization of the resulting
         * log lines. For example, they can be used to change the level key name or to enrich the default metadata.
         */
        formatters?: {
            /**
             * Changes the shape of the log level. The default shape is `{ level: number }`.
             * @param label The label of the level (e.g. `info`).
             * @param number The numeric value (e.g. `30`).
             * @example
             * level (label, number) {
             *  return { level: number }
             * }
             */
            level?: (label: string, number: number) => object;
            /**
             * Changes the shape of the bindings. The default shape is `{ pid, hostname }`. It will be called every time
             * a child logger is created.
             * @param bindings The bindings object.
             * @example
             * bindings (bindings) {
             *  return { pid: bindings.pid, hostname: bindings.hostname }
             * }
             */
            bindings?: (bindings: Bindings) => object;
            /**
             * Changes the shape of the log object. This function will be called every time one of the log methods (such
             * as `.info`) is called. All arguments passed to the log method, except the message, will be pass to this
             * function. By default it does not change the shape of the log object.
             * @example
             * log (object) {
             *  return object
             * }
             */
            log?: (object: object) => object;
        };
```

The main difference is that `formatters` does not seem to accept arbitrary keys, but instead has three preset keys. I'm not sure about using `object` for the inputs and outputs of these functions, but it seems consistent with other types in this file.

Reply:
I suggest to change `level?: (label: string, number: number) => object;` to `level?: (label: string, number: number) => Record<string, string|number>;`

based on example here : https://github.com/pinojs/pino/blob/46911703fca657a44419f9abba2fe94db2e0e650/pino.js#L179...L193

Reply:
Thanks for the input. It looks like the functions you linked to are the default implementations provided by Pino. The docs say that `{level: number}` is the _default_ shape, not necessarily the only supported shape, so I think it's safest to use `object` as a return type here in order to avoid breaking anyone's workflow.